### DISCUSSION TITLE
AutoKey Dependencies Need To Be Sorted Out

### DISCUSSION THREAD

**Discussion by Elliria** [autokey/autokey#978] [Maintenance goals]
It seems that AutoKey's dependencies are in a bit of disarray. I suspect that this may be the cause of some issues. Rather than putting them into this post, because the information is too bulky, I put them into a semicolon-separated CSV file that gives a somewhat-dramatic visual of what's under the hood of AutoKey:

[AutoKeyDependencyLocations.csv](https://github.com/user-attachments/files/17981438/AutoKeyDependencyLocations.csv)

It shows:
* what's where
* what's not where
* what's older
* what overlaps
* etc.

**Comment by josephj11** [comment]
I got the fie to work, but it is delimited by semicolons and CSV implies comma separated values. The different delimiter works, but needs to be specified. With quoted fields, comma delimiters should work even if some fields contain embedded commas.

That is a big table!

**Comment by Elliria** [comment]
I specified it, unless you mean it has to be specified each time the file is loaded (which it does). Either way, it's a text file, so you could edit it and do a search-and-replace on the semicolons.

Does the data make any sense to you? It's a jumbled mess from where I sit.

**Comment by josephj11** [comment]
I missed the fine print, LOL. Sorry. At a glance, it looked fine, but there was enough of it to make me skip actually trying to understand it. If you are working on it, I can put in the effort to actually try to parse it in detail...

Once I got it to read successfully, I saved it in spreadsheet (.ods) format in Libreoffice, so it opens correctly by default.

**Comment by Elliria** [comment]
That's a good idea. In a spreadsheet, you can turn the top row into a clickable header-row so that the data can be sorted in a variety of ways.

My motive for creating this discussion is to get help in understanding the data so that we can make solid decisions about where future things ought to be and not be.

**Reply by josephj11** [reply]
I guess that means I'm going to have to study it. ;)
